app/Http/Controllers/QueryController.php

```diff
@@ -157,4 +162,57 @@ public function membershipTotals(Request $request)
             ]);
         }
     }
+
+    public function activityStatistics(Request $request)
+    {
+
+        if ($request->missing('start') || $request->missing('end')) {
+            if (intval(date('n')) >= 9) {
+                $year_start = intval(date('Y'));
+            } else {
+                $year_start = intval(date('Y')) - 1;
+            }
+            $start = strtotime("$year_start-09-01 00:00:01");
+            $end = date('U');
+        } else {
+            $start = strtotime($request->start);
+            $end = strtotime($request->end) + 86399; // Add one day to make it inclusive.
+        }
+
+        $eventCategories = EventCategory::withCount(['events' => function ($query) use ($start, $end) {
+            $query->where('start', '>=', $start)->where('end', '<=', $end);
+        }])->get()->sortBy('name');
+
+        foreach ($eventCategories as $category) {
+            $category->spots = Activity::whereHas('event', function ($query) use ($category, $start, $end) {
+                $query->where('category_id', $category->id)->where('start', '>=', $start)->where('end', '<=', $end);
+            })->where('participants', '>', 0)
+                ->sum('participants');
+
+            $category->signups = ActivityParticipation::whereHas('activity', function ($query) use ($category, $start, $end) {
+                $query->whereHas('event', function ($query) use ($category, $start, $end) {
+                    $query->where('category_id', $category->id)->where('start', '>=', $start)->where('end', '<=', $end);
+                })->where('participants', '>', 0);
+            })->count();
+
+            $category->attendees = Activity::where('participants', '>', 0)->whereHas('event', function ($query) use ($category, $start, $end) {
+                $query->where('category_id', $category->id)->where('start', '>=', $start)->where('end', '<=', $end);
+            })->sum('attendees');
+        }
+
+        $events = DB::table('events')
```

Comment:
Why use raw SQL here instead of Eloquent query builders?

Reply:
Because the dates are in the wrong format in the DB, and the only way to group by what I want is to use the built-in sql YEAR() and MONTH() functions.

Reply:
Use the model and then use selectRaw to do your magic sql stuff. Like Event::query()->selectRaw('year(created_at) as year')->groupBy('year')->get()